### DISCUSSION TITLE
Purpose of `IConditionalOperation.IsRef`

### DISCUSSION THREAD

**Discussion by YoshiRulz** [dotnet/roslyn#75107] [Q&A] [answered]
The docs simply say:
https://github.com/dotnet/roslyn/blob/214d43a8c5a3880c69ae390c0eb7cdd33dbf0e17/src/Compilers/Core/Portable/Operations/OperationInterfaces.xml#L1138-L1142
but, for C# ternaries at least, it doesn't seem to be related to whether the inferred type of the expression is a reference type.
I see that it was introduced in #22933 with some `ref T` stuff, so I'm guessing it's instead related to that.

**Comment by 333fred** [comment, marked as answer]
It's for ref ternaries. See https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator#conditional-ref-expression.
